src/pages/SignUpForm.jsx

```diff
@@ -1,71 +1,116 @@
-import React from 'react';
-import shape1 from './assets/shape1.png';
+//src/pages/SignUpForm.jsx
+import React, { useState } from 'react';
+import { useRegister } from '../hooks/useRegister';
+import shape1 from '../assets/shape1.png';
+import DOMPurify from 'dompurify';
 
 const SignUpForm = () => {
+  const [formData, setFormData] = useState({
+    name: '',
+    email: '',
+    password: '',
+    password_confirmation: '',
+  });
+  const { register, isLoading, error } = useRegister();
+
+  const handleChange = (e) => {
+    setFormData({ ...formData, [e.target.name]: e.target.value });
+  };
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    if (formData.password !== formData.password_confirmation) {
+      alert('Passwords do not match');
```

Comment:
it works for testing, but for production app, you almost never see errors in alerts. You can add div with error and show it or hide